// !DIAGNOSTICS: -UNUSED_VARIABLE, -UNUSED_PARAMETER
class A

fun <!UNRESOLVED_REFERENCE!>@x<!> A.foo(a: <!UNRESOLVED_REFERENCE, WRONG_ANNOTATION_TARGET!>@x<!> Int) {
    val v: <!UNRESOLVED_REFERENCE, WRONG_ANNOTATION_TARGET!>@x<!> Int = 1
}